### DISCUSSION TITLE
Derived Fixture Behavior

### DISCUSSION THREAD

**Discussion by terryaney** [xunit/xunit#2894] [Question]
If I have the following setup:

```
public class CalcEngineFactory : WebApplicationFactory<IApiMarker>
{
	// Code
}

public class CalcEngineReadFactory : CalcEngineFactory
{
	// Code
}

public class UpdateTests : IClassFixture<CalcEngineFactory>
{
	private readonly CalcEngineFactory factory;
	public UpdateTests( CalcEngineFactoryfactory ) => this.factory = factory;
}
public class ReadTests : IClassFixture<CalcEngineReadFactory>
{
	private readonly CalcEngineReadFactory factory;
	public ReadTests( CalcEngineReadFactory factory ) => this.factory = factory;
}
```

Will I get two 'instances' made?  When I 'run all' tests, I always get tons of failures in my `UpdateTests`, but if I go to that class and click the 'Run All Tests' code lens, they succeed.  Unfortunately I can't 'debug all tests' (read and update) that I know how to try to ensure this.  But thinking there must be some misunderstanding on my part about how this is working.  Do I need a separate `CalcEngineUpdateFactory` class to make it work how I want?

Digging more to see if other issues as well.

**Comment by bradwilson** [comment]
> Will I get two 'instances' made?

Yes: one instance of `CalcEngineFactory` for the tests in `UpdateTests`, and one instance of `CalcEngineReadFactory` for the tests in `ReadTests`.

> When I 'run all' tests, I always get tons of failures in my UpdateTests

Without knowing more about your actual situation, I can't really begin to guess what the problems are. Maybe my answer above helps you figure it out...? If not, you can try and create a repro project that I can look at that illustrates your issue.

**Reply by terryaney** [reply]
Thanks.  It was my error (of course), but nice to rule out derived fixture/factory concern.  Not xUnit related but guessing you have guess.  I've been looking for best practice when leveraging Docker containers and TestContainers.Sql.  Originally, I had about 11-12 `IClassFixtures` (same number of containers launched) and figured I should minimize that.  So I made a `ICollectionFixture` and my tests ran about 2-2.5x slower.  Obviously because of synchronous nature with `ICollectionFixture`.  Any advice there?  I've refactored back to where I started basically (but limiting it to 8 containers) and am in a pretty good spot I guess.

**Reply by bradwilson** [reply]
You are correct that using collection fixtures disabled parallelism, which could make the run time seem slower overall.

This is the open issue for v3 to allow more control over parallelism with shared context: https://github.com/xunit/xunit/issues/2055
